src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/RejectionException.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptions;
+
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.ErrorCode;
+
+public abstract class RejectionException extends RuntimeException {
+
+    // might contain sensitive data
+    private final String detailMessage;
+    private final ErrorCode errorCode;
+
+    public RejectionException(ErrorCode errorCode, String message) {
+        super(message);
+        this.detailMessage = message;
+        this.errorCode = errorCode;
+    }
```

Comment:
This one might be a bit risky, because developers might not see the distinction between `message` and `detailMessage`. Maybe we could drop this constructor?

Reply:
if we drop we need to change all the other subclasses. for now we can keep like this and add comment.

Reply:
ok